src/main/java/uk/gov/hmcts/reform/sscs/services/TransformationService.java

```diff
@@ -60,18 +60,33 @@
         }
         List<AppealCase> appealCases = result.getAppealCases().getAppealCaseList();
 
-        return (appealCases == null) ? Collections.emptyList() : appealCases.stream()
-            .filter(c -> c.getCreateDate() != null && ignoreCasesBeforeDate.isBefore(c.getCreateDate()))
-            .filter(c -> {
-                if (c.getAppealCaseNino() == null) {
-                    log.debug("NINO({}) is null for case number({}):", c.getAppealCaseNino(), c.getAppealCaseRefNum());
-                    return false;
-                }
-                return !ROBOTIC_NINO_FOR_TESTING_PURPOSE.equalsIgnoreCase(c.getAppealCaseNino()
-                    .replaceAll("\\s+", ""));
-            })
+        return (appealCases == null) ? Collections.emptyList() : doTransformGapsCasesToSscsCases(appealCases);
+    }
+
+    private List<SscsCaseData> doTransformGapsCasesToSscsCases(List<AppealCase> appealCases) {
+        return appealCases.stream()
+            .filter(c -> isCaseCreatedDateBeforeThan(c.getCreateDate()))
+            .filter(c -> !isNinoNull(c.getAppealCaseNino(), c.getAppealCaseRefNum()))
+            .filter(c -> !isAroboticTestCase(c))
```

Comment:
robotic check logic is part of isNinoNull, but here we have added extra filter which I think its changes the logic